core/src/main/java/org/elasticsearch/search/controller/SearchPhaseController.java

```diff
@@ -191,19 +193,39 @@ private static long optionalSum(long left, long right) {
                 offset = 0;
             }
             ScoreDoc[] scoreDocs = result.topDocs().scoreDocs;
+            ScoreDoc[] docs;
             if (scoreDocs.length == 0 || scoreDocs.length < offset) {
-                return EMPTY_DOCS;
-            }
-
-            int resultDocsSize = result.size();
-            if ((scoreDocs.length - offset) < resultDocsSize) {
-                resultDocsSize = scoreDocs.length - offset;
+                docs = new ScoreDoc[0];
+            } else {
+                int resultDocsSize = result.size();
+                if ((scoreDocs.length - offset) < resultDocsSize) {
+                    resultDocsSize = scoreDocs.length - offset;
+                }
+                docs = new ScoreDoc[resultDocsSize];
+                for (int i = 0; i < resultDocsSize; i++) {
+                    ScoreDoc scoreDoc = scoreDocs[offset + i];
+                    scoreDoc.shardIndex = shardIndex;
+                    docs[i] = scoreDoc;
+                }
             }
-            ScoreDoc[] docs = new ScoreDoc[resultDocsSize];
-            for (int i = 0; i < resultDocsSize; i++) {
-                ScoreDoc scoreDoc = scoreDocs[offset + i];
-                scoreDoc.shardIndex = shardIndex;
-                docs[i] = scoreDoc;
+            Suggest suggest = result.queryResult().suggest();
+            if (suggest != null) {
+                int numSuggestDocs = 0;
+                for (CompletionSuggestion suggestion : suggest.filter(CompletionSuggestion.class)) {
```

Comment:
oops, thanks for the suggestion. I think the restructuring made the code simpler